test/unit/service_test.rb

```diff
@@ -391,21 +391,20 @@ class DestroyingServiceTest < ActiveSupport::TestCase
     end
 
     test 'destroying service creates a related event' do
-      service          = FactoryBot.create(:service)
-      service_plan     = FactoryBot.create(:service_plan, issuer: service)
```

Comment:
We already have another test to check that its plans are deleted:
https://github.com/3scale/porta/blob/13df75e504e401679efb0fe05086d7260cf7bb5e/test/unit/service_test.rb#L364-L373
And when it comes to the contracts, that's responsibility of the plan, not of the service 😄 And I don't know if we have a test for it but I will check on Monday 😉